### ISSUE TITLE
Remove inverse value when resource value is removed

### ISSUE THREAD

**Issue by jimsafley** [omeka-s-modules/InverseProperties#3]
There is interest in a way to automatically remove an inverse value when a resource value is removed. See [this forum comment](https://forum.omeka.org/t/inverse-properties/20537/10):

> When an already “linked” property is removed (I don’t mean when an item is deleted, just the specific property) it should be removed also from any other connected item / item-set. I understand that technically this is complicated and would demand to parse all items / items-sets so I am not sure if it’s a good idea after all… maybe an option in the module’s settings to be switched on/off depending on the case

This is tricky because the module does not track inverse values, nor is there a reliable way to track individual values given the way Omeka handles value hydration. So it wouldn't be a simple matter of detecting a value has been removed and referencing a tracking table. This leaves removing by a heuristic, parsing one value at a time, the only solution, which would make *every single* update an expensive process.

**Comment by itsam** [comment]
You are right that such an action applied on every single update or insert is a very expensive process. One idea is to have something similar to https://omeka.org/s/modules/TypesenseSearch/ module. In this module you are able to force a re-index to the typesense server at any time in the background. 
So, similarly, users will be able to trigger a "Clean up" process (probably a cron job) which will do all the heavy process outside the usual flow. It's not ideal but definitely will be better than having to do it completely manually...
Just my 2 cents on this...

**Comment by jimsafley** [comment]
Interesting, though removing inverse values must be done at the time of value removal. Otherwise there would be no way to know which values were, at one point, inverse values. I suppose the module could detect that an inverse value was removed, cache the  relevant IDs (resource, inverse resource, property), and provide a button somewhere that runs a background job that removes previously unlinked inverses. In fact, that's probably similar to what you're recommending.

Even so, there is some risk of data loss in doing this asynchronously from the initial removal. As time goes on and resources become further described, inverse resources may have values that just happen to line up with the heuristic, and they get removed mistakenly. Not a likely scenario for most, but something to consider.

I don't plan on doing this any time soon, but will look into it at a future date. There may still be a way to do this synchronously without too much effort or processing.
